src/plugins/data/public/ui/filter_bar/filter_item.tsx

```diff
@@ -62,57 +73,47 @@ class FilterItemUI extends Component<Props, State> {
       this.togglePopover();
     }
   };
-  public render() {
-    const { filter, id } = this.props;
-    const { negate, disabled } = filter.meta;
-    let hasError: boolean = false;
 
-    let valueLabel;
-    try {
-      valueLabel = getDisplayValueFromFilter(filter, this.props.indexPatterns);
-    } catch (e) {
-      getNotifications().toasts.addError(e, {
-        title: this.props.intl.formatMessage({
-          id: 'data.filter.filterBar.labelErrorMessage',
-          defaultMessage: 'Failed to display filter',
-        }),
-      });
-      valueLabel = this.props.intl.formatMessage({
-        id: 'data.filter.filterBar.labelErrorText',
-        defaultMessage: 'Error',
-      });
-      hasError = true;
-    }
-    const dataTestSubjKey = filter.meta.key ? `filter-key-${filter.meta.key}` : '';
-    const dataTestSubjValue = filter.meta.value ? `filter-value-${valueLabel}` : '';
-    const dataTestSubjDisabled = `filter-${
-      this.props.filter.meta.disabled ? 'disabled' : 'enabled'
-    }`;
-    const dataTestSubjPinned = `filter-${isFilterPinned(filter) ? 'pinned' : 'unpinned'}`;
+  private isValidLabel(labelConfig: LabelOptions) {
```

Comment:
I broke down the logic in this component to helper functions for easier debugging and reviewing